x-pack/plugins/ml/server/lib/alerts/alerting_service.ts

```diff
@@ -345,67 +406,30 @@ export function alertingServiceProvider(mlClient: MlClient, esClient: Elasticsea
       jobIds
     );
 
-    const result = response.body.aggregations as {
-      alerts_over_time: {
-        buckets: Array<
-          {
-            doc_count: number;
-            key: number;
-            key_as_string: string;
-          } & {
-            [key in PreviewResultsKeys]: {
-              doc_count: number;
-            } & {
-              [hitsKey in TopHitsResultsKeys]: {
-                hits: { hits: any[] };
-              };
-            };
-          }
-        >;
-      };
-    };
+    const result = response.body.aggregations;
 
     const resultsLabel = getAggResultsLabel(params.resultType as AnomalyResultType);
 
-    return (
-      result.alerts_over_time.buckets
-        // Filter out empty buckets
-        .filter((v) => v.doc_count > 0 && v[resultsLabel.aggGroupLabel].doc_count > 0)
-        // Map response
-        .map((v) => {
-          const aggTypeResults = v[resultsLabel.aggGroupLabel];
-          const requestedAnomalies = aggTypeResults[resultsLabel.topHitsLabel].hits.hits;
-
-          const topAnomaly = requestedAnomalies[0];
-          const alertInstanceKey = getAlertInstanceKey(topAnomaly._source);
+    const formatter = getResultsFormatter(params.resultType as AnomalyResultType);
```

Comment:
`params.resultType` is being cast as `AnomalyResultType` a lot of times in this file. can the `MlAnomalyDetectionAlertParams` type be updated to make this correct?

Reply:
Fixed in https://github.com/elastic/kibana/pull/93939/commits/d15f9fdee4554788985fafa25c916d1135759ed5